### ISSUE TITLE
Add an SNS notification channel for increase / decrease throughput?

### ISSUE THREAD

**Issue by awinder** [channl/dynamodb-lambda-autoscale#44]
If dynamodb-lambda-autoscale had the ability to fire off an SNS notification for every increase or decrease of capacity, it could help with building some tooling around this.  For instance, i'd love to create a simple Slackbot that just notifies when capacity increases/decreases, so that I can keep an eye on what is happening and when.  Thoughts?

**Comment by tmitchel2** [comment]
I think this is more analytics which i think should be separated out from the actual autoscaler itself.  You might be able to set SNS notification based upon a cloudwatch trigger without changes to this.

**Comment by awinder** [comment]
Yeah that could work, but then I'd miss out / have to replicate logic from this to know _why_ a decision was made to tune up / down from the autoscaler (as would anyone else using this project).  I guess I could also subclass the `ProvisionerConfigurableBase` class but, then I'd have to maintain a fork

**Comment by tmitchel2** [comment]
Other people have mentioned the requirement of forking as a concern which i can understand and agree with, however i don't want this package to turn into a big blob.  I think i will end up making it so you auto generate a template project, pull down this function using npm and then configure / add plugins to do additional work.  That way users will be able to create and share additional features without polluting the 'small' core functionality.  This will take a little time but it will definitely happen...

**Comment by awinder** [comment]
Ah that makes perfect sense, thanks for explaining the intentions.  I will keep an eye out.  If you get to the point where you have a solid arch. vision for some of this work and just need an extra set of hands, shoot me a message 😃

**Comment by alexraju91** [comment]
@awinder I'm having a similar requirement, to get the update notification in slack. Were you able to complete this.

**Comment by awinder** [comment]
Yeah I did, you can check out my fork if it's helpful https://github.com/awinder/dynamodb-lambda-autoscale/.  I'm not actively maintaining anything with this though; we ended up going forward with dynamic-dynamodb instead which supported SNS messaging out of the box

**Comment by alexraju91** [comment]
This looks great. I will try your code and let you know. Thank you !!
